backend/oasst_backend/api/v1/login.py

```diff
@@ -0,0 +1,68 @@
+import requests
+from fastapi import APIRouter, Depends, HTTPException
+from oasst_backend import auth
+from oasst_backend.api import deps
+from oasst_backend.config import Settings
+from oasst_backend.models import Account
+from oasst_shared.exceptions.oasst_api_error import OasstError, OasstErrorCode
+from oasst_shared.schemas import protocol as protocol_schema
+from sqlmodel import Session
+from starlette.status import HTTP_401_UNAUTHORIZED
+
+router = APIRouter()
+
+# TODO: replace placeholder
+REDIRECT_URI = "http://localhost:8000/api/v1/callback_discord"
```

Comment:
For callback URLS, I'd recommend the pattern `/api/v1/callback/{provider}`.

Then later we can drop in other provides as we develop them: `email`, `github` etc.

This seems to be the URL pattern NextAuthJS uses.